### ISSUE TITLE
enter your clash account name: this account seems to not exist

### ISSUE THREAD

**Issue by spock** [dynamicbk/ClashBotDiscord#73]
So I've managed to get the bot running.
It only exposed 3 commands to me, of which `!start` looked promising.

However, when the bot asked
> Great! Let's link your accounts now.
> Please enter your clash account name:

and I've responded with name (and then tag),
all I got back was

> This account seems to not exist
> Do you want to try entering this account again? If you cannot find this account, [snip]

I guess the database needs to be pre-populated first? So datamanager-initialize isn't sufficient?

Creating `developers` role and adding myself to it revealed more commands, but even using `!linkmemberaccountbytag` resulted in the same `This account seems to not exist`...

A potentially related observation is that I'm getting too many `first load` messages:
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: starting database watcher
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: database watcher looping
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:29:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: database watcher looping
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:30:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: database watcher looping
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load
> 2020-01-02 00:24:31:DEBUG:ClashBotDiscordBot: first load

**Comment by dynamicbk** [comment]
The READMEs are in need of some love, I'll take a look at getting them cleaned up shortly for better guidance.

That said, you're close.

A quick overview is that you need to do this:

1) create a directory to work out of, where you will run all commands, where saved data will go, as well as where the database will live
2) run clashbot-data-manager-initialize
3) run clashbot-data-manager (this is the step I think you aren't doing)
4) run clashbot-discord-bot

both steps 3 and 4 are long running processes- the data manager downloads, saves, and processes the data, while the discord bot just reads the data from the database. I usually leave each one running in a separate screen instance.

**Comment by spock** [comment]
indeed, step 3 is what I wasn't doing at all.
thanks, will give it a try! 

I've changed looping sleep() statements to 60 seconds from 1 and 5 for different watchers - but now I understand that this will likely affect only reading from the db, so that's ok to leave them as-is. 

I've found a few missing config keys in example configs, should I sent a merge request? unfortunately, I've ran black on the code first.... 😁

**Comment by dynamicbk** [comment]
Any chance you could pull those changes into a different branch? I imagine the changes from black will result in a pretty long diff :)

Let me know of any other issues as you’re setting this up, I haven’t set it up from scratch in a while. I did just create a new issue to improve the startup story for the discord bot, similar to the data manager initializer. 

Btw, the first time the data manager runs, it’ll check for already saved data files, so the very first load will be slower than future ones. Once it’s read it’s first batch of data, it’ll be much faster.

**Comment by spock** [comment]
Small request sent!

I've spent a while figuring stuff out :)

So far:
- how to get discord server/channel numeric ID: https://support.discordapp.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-
- getting supercell developer API token: https://developer.clashofclans.com/
- how to create a new discord bot, and then later invite it: https://discordpy.readthedocs.io/en/latest/discord.html (manual), https://discordapp.com/developers/applications/ (dashboard)

Still a mystery and/or unfinished:
- what will the bot keep track of? (I'm interested in per-player CW stats, donations, and clan games scores.)
- generally, what is the flow of operation? (I somewhat understand what `!start` does, but that's it)
- which roles do I need? how automatic roles assignment works, and would it conflict with Minion bot?
- which channels are required for what? the list of channels in config is quite long...
- setting up S3/CloudWatch with correct permissions and zones; I've seen us-east-1 hardcoded somewhere, and I've selected Ohio, so I hope that will still work :)
